infrastructure/src/main/java/org/corfudb/infrastructure/BatchWriterOperation.java

```diff
@@ -1,37 +1,32 @@
 package org.corfudb.infrastructure;
 
-import java.util.List;
 import java.util.concurrent.CompletableFuture;
 
 import lombok.Data;
 
-import org.corfudb.protocols.wireprotocol.LogData;
+import org.corfudb.protocols.wireprotocol.CorfuPayloadMsg;
 
 /**
- * Created by maithem on 11/28/16.
+ * This container is used by the logunit to add work to the batch writer. Its also used
+ * to coordinate job completion.
  */
-
 @Data
-public class BatchWriterOperation {
+public class BatchWriterOperation<T> {
     public enum Type {
         SHUTDOWN,
         WRITE,
         RANGE_WRITE,
+        TRIM,
         PREFIX_TRIM,
         SEAL,
         RESET,
         TAILS_QUERY
     }
 
     private final Type type;
-    private final Long address;
-    private final LogData logData;
-    private final Long epoch;
-    private final List<LogData> entries;
-    private final CompletableFuture future;
-    private Exception exception;
-
+    private final CorfuPayloadMsg msg;
+    private Object retVal;
```

Comment:
Why object? Can we add a generic parameter which specifies retval type?

Reply:
We can, but I see it unnecessary at the moment because its just a placeholder. In other words, we don't have code that casts it.

Reply:
Even if we don't have casts I still vote for a prev version. `Object` confuses when reading code

Reply:
Done.